pkg/cvo/status.go

```diff
@@ -413,6 +429,53 @@ func updateClusterVersionStatus(cvStatus *configv1.ClusterVersionStatus, status
 	}
 }
 
+// getReasonMessageFromError returns the reason and the message from an error.
+// If the reason or message is not available, an empty string is returned.
+func getReasonMessageFromError(err error) (reason, message string) {
+	if uErr, ok := err.(*payload.UpdateError); ok {
+		reason = uErr.Reason
+	}
+	if err != nil {
+		message = err.Error()
+	}
+	return reason, message
+}
+
+// filterErrorForFailingCondition filters out update errors based on the given
+// updateEffect from a MultipleError error. If the err has the reason
+// MultipleErrors, its immediate nested errors are filtered out and the error
+// is recreated. If all nested errors are filtered out, the original
+// MultipleError error is returned, as to keep the internal logic unchanged.
```

Comment:
"as to keep the internal logic unchanged."
I have to admit that this is not a clear comment to me, i.e., I do not get the reason why we implement it that way.

Let me ask this from another angle:
If we call the function `filterErrorForFailingCondition` with an arg different from `payload.UpdateEffectNone`, do we still want the logic like that?

Reply:
Yes, the comment does not make sense in the context. Let me rethink the logic and the wording.

Reply:
The PR dances on a thin line between fixing what's wrong and leaving the existing logic as is to not cause a regression and keep the commit simple. The original reasoning was to leave the current logic functioning as is but filter out unimportant messages **only** if there is at least one valid message in the `MultipleErrors` message to fix the current bug.

However, right now, I can't think of a valid reason to continue doing this and not filtering all `MultipleErrors` immediate `UpdateEffectNone` nested errors. This is a good indicator that the change is not needed.

Reply:
Feel free to check a proposal in a new draft commit https://github.com/openshift/cluster-version-operator/pull/1050/commits/8a5567a51e3d4e0b52678196414650b757eb5634

Reply:
To me, it looks reasonable to treat `UpdateEffectNone` not en error at all.
However, I know too little to see all the impacts it would bring.  :)
I would give another one the chance to take a look before I comment with LGTM.